bokun.yaml

```diff
@@ -5537,413 +5734,535 @@ definitions:
+      productCategory:
+        $ref: '#/definitions/ProductCategoryEnum'
+        description: 'The product category, will be one of: ACTIVITIES, ACCOMMODATION, TRANSPORT, CAR_RENTALS'
+      box:
         type: boolean
-        description: 'Specifies if there is a particular dress code that the participants need to adhere to. The type of dress code should be described in text items.'
-      passportRequired:
+        description: 'Specifies whether this is a boxed version of another product.'
+      inventoryLocal:
         type: boolean
-        description: 'Whether passenger needs to bring their passport for this activity.'
-      ticketPerPerson:
+        description: 'Specifies if this activity inventory is managed locally or it is external inventory service product. False if this activity is external inventory service product, true - otherwise.'
+      inventorySupportsPricing:
         type: boolean
-        description: 'Whether ticket is generated per person or per booking.'
-      supportedAccessibilityTypes:
-        type: array
-        description: 'Specifies whether the activity is suitable for individuals with accessibility issues.'
-        items:
-          $ref: '#/definitions/AccessibilityType'
-      startPoints:
-        type: array
-        description: 'Only relevant if meetingType == MEET_ON_LOCATION or MEET_ON_LOCATION_OR_PICK_UP. List of all places where this activity can start or be joined.'
-        items:
-          $ref: '#/definitions/Place'
-      guidanceTypes:
-        type: array
-        description: 'The list of types of guidance on offer, along with what languages each is offered in.'
-        items:
-          $ref: '#/definitions/Guidance'
-      activityCategories:
-        type: array
-        description: 'The list of categories that the tour/activity/attraction falls into.'
-        items:
-          $ref: '#/definitions/ActivityCategory'
-      activityAttributes:
-        type: array
-        description: 'Attributes that describe the tour/activity/attraction.'
-        items:
-          $ref: '#/definitions/ActivityAttribute'
-      actualVendor:
+        description: 'If this activity is external inventory service product, then equals to SUPPORTS_PRICING of the connected plugin.'
+      inventorySupportsAvailability:
+        type: boolean
+        description: 'If this activity is external inventory service product, then equals to SUPPORTS_AVAILABILITY of the connected plugin.'
+      creationDate:
+        type: string
+        description: 'The date the activity was first created.'
+      lastModified:
+        type: string
+        description: 'The date that the activity content was last modified.'
+      lastPublished:
+        type: string
+        description: 'The date that the activity was last changed and published.'
+      published:
+        type: boolean
+#        description: 'NEED MORE INFO!, as could not find when it is true. Unpublished product is not returned in response.'
```

Comment:
The general rule of programming is that a "small thing" (in this case this attribute) should not be aware of external context (in this case - when is it used). Just make description sound that this reflects if the product is published, and since the concept of publishing is understood by users of our system, they will get it. Because if we mention that this is "always true" and if we add another endpoint which uses this DTO and which returns unpublished activities, this documentation will suddenly become incorrect.

Reply:
"published" is a property in ProductDto. There is method in this class setPublished(boolean), however it is not used in code anywhere. So, could not understand if it gets another value - it is always "false"(and actually, as we get only published products via API, so "false" value does not correspond to reality also).

Reply:
Recommendation: File bug - always true or remove from response.